src/NativeScript/inlineFunctions.js

```diff
@@ -14,6 +14,103 @@ Object.assign(global, {
     NSMakeRange(location, length) {
         return new NSRange({ location, length });
     },
+
+    // Decorators support    
+    __decorate(decorators, target, key, desc) {
+        var c = arguments.length;
+        var r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc;
+        var d;
+
+        if (typeof global.Reflect === "object" && typeof global.Reflect.decorate === "function") {
+            r = global.Reflect.decorate(decorators, target, key, desc);
+        }
+        else {
+            for (var i = decorators.length - 1; i >= 0; i--) {
+                if (d = decorators[i]) {
+                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
+                }
+            }
+        }
+        return c > 3 && r && Object.defineProperty(target, key, r), r;
+    },
+    __param(paramIndex, decorator) {
+        return function (target, key) {
+            decorator(target, key, paramIndex);
+        }
+    },  
+
+    // Decorators  
+    ObjC(name) {
+        var additionalParams = [];
+        var hasName = (name !== undefined && typeof name === "string");
+        var argsShift = hasName ? 1 : 0;
+        for (var i = argsShift; i < arguments.length; i++) {
+            additionalParams[i - argsShift] = arguments[i];
+        }
+
+        return function (target, propertyKey, descriptor) {
+            if (propertyKey === undefined) { // Class decorator
+                var implementation = {};
+                var baseTarget = Object.getPrototypeOf(target);
+                
+                for (var prop in target.prototype) {
+                    if (target.prototype.hasOwnProperty(prop)) {
+                        implementation[prop] = target.prototype[prop];
+                    }
+                }
+
+                var newName = hasName ? name : target.name;                
+                var extended = baseTarget.extend(implementation, {
```

Comment:
@ivanbuhov  Agreed. 

@PanayotCankov , for overloading, the problem is not with the implementation of it. It is already working fine. The problem is with the definitions, and mainly comes because the `returnType` for the method and protocols optional parameters are any. So in the d.ts the definitions conflict with each other. For example:
```ts
function ObjC(returnType: any): MethodDecorator;
function ObjC(...protocols: any[]): ClassDecorator;
```
These will not work. You have to define them as:
```ts
function ObjC(type: any): ClassDecorator & MethodDecorator;
function ObjC(...protocols: any[]): ClassDecorator;
```
But then the help provided in the tooltip will not be very clear because `type` for the class will actually be a protocol and for the method this will be the return type.